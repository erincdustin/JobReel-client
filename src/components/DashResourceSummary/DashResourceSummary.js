import React from 'react';
import { Link } from 'react-router-dom';
import './DashResourceSummary.css';
import jobReelContext from '../../context/JobReelContext';
import jobReelApiService from '../../services/jobreel-api-service';

class DashResourceSummary extends React.Component {

  static contextType = jobReelContext

  componentDidMount() {
    if (this.context.savedJobs.length === 0) {
      jobReelApiService.getSavedResources()
        .then(res => {
          this.context.setResources(res);
        })
    }
  }

  renderResourceSummaries = () => {
    let resources = Array.from(this.context.resources);
    resources = resources.sort((a, b) => {
      return new Date(b.date_added) - new Date(a.date_added);
    });
    resources = resources.slice(0, 3).map(resource => {
      return (
        <li key={resource.resource_id}>{resource.type.toUpperCase()} - {resource.title}</li>
      )
    })
    return (
      <ul className='ul'>
        {resources}
      </ul>
    )
  }
  
  render() {
    return (
      <>
        <div className="DashResourceSummary">
          <h4>{this.props.section}</h4>
        </div>
        <div className='options'>
            {this.renderResourceSummaries()}
        </div>
        <div className='viewAll'>
          <Link to="/resources">View all</Link>
        </div>
        
      </>
    )
  }
}

export default DashResourceSummary;