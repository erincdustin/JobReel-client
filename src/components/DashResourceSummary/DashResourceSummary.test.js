import React from 'react';
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import PropTypes from 'prop-types'
import { BrowserRouter } from 'react-router-dom'
import DashResourceSummary from './DashResourceSummary'

class ContextProvider extends React.Component{
  static childContextTypes = {
    myContext: PropTypes.object,
  }

  getChildContext = () => ({
    myContext: {
      resources: [1, 2, 3]
    }
  })

  render(){
    return this.props.children
  }
}

describe('DashResourceSummary smoke test', () => {
  it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<BrowserRouter><DashResourceSummary /></BrowserRouter>, div);
      ReactDOM.unmountComponentAtNode(div);
  })
})

describe('DashResourceSummary snapshot test', () => {
  it('renders the UI as expected', () => {
    const wrapper = shallow(<ContextProvider><DashResourceSummary /></ContextProvider>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})