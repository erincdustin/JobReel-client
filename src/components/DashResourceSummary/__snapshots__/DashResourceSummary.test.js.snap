// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashResourceSummary snapshot test renders the UI as expected 1`] = `<DashResourceSummary />`;
